tachyon/zk/plonk/examples/fibonacci/fibonacci1_circuit.h

```diff
@@ -0,0 +1,172 @@
+#ifndef TACHYON_ZK_PLONK_EXAMPLES_FIBONACCI_FIBONACCI1_CIRCUIT_H_
+#define TACHYON_ZK_PLONK_EXAMPLES_FIBONACCI_FIBONACCI1_CIRCUIT_H_
+
+#include <array>
+#include <memory>
+#include <utility>
+#include <vector>
+
+#include "tachyon/zk/plonk/constraint_system/circuit.h"
+
+namespace tachyon::zk::plonk {
+
+template <typename F>
+struct AssignedRow {
+  AssignedCell<F> a;
+  AssignedCell<F> b;
+  AssignedCell<F> c;
+
+  AssignedRow() = default;
+  AssignedRow(AssignedCell<F>&& a, AssignedCell<F>&& b, AssignedCell<F>&& c)
+      : a(std::move(a)), b(std::move(b)), c(std::move(c)) {}
+};
+
+template <typename F>
+struct Fibonacci1Config {
+  using Field = F;
+
+  Fibonacci1Config(std::array<AdviceColumnKey, 3> advice, Selector selector,
+                   const InstanceColumnKey& instance)
+      : advice(std::move(advice)), selector(selector), instance(instance) {}
+
+  Fibonacci1Config Clone() const {
+    return Fibonacci1Config(advice, selector, instance);
+  }
+
+  std::array<AdviceColumnKey, 3> advice;
+  Selector selector;
+  InstanceColumnKey instance;
+};
+
+template <typename F>
+class Fibonacci1Chip {
+ public:
+  explicit Fibonacci1Chip(Fibonacci1Config<F>&& config)
+      : config_(std::move(config)) {}
```

Comment:
Since it doesn't have movable members, I think below makes more sense to me.

```suggestion
  explicit Fibonacci1Chip(const Fibonacci1Config<F>& config)
      : config_(config) {}
```